core/tests/unit/test_cicd_pipeline_stack.py

```diff
@@ -219,3 +219,233 @@ def test_cicd_pipeline_simple(cdk_app: App) -> None:
             ),
         },
     )
+
+
+def test_cicd_pipeline_full(cdk_app: App) -> None:
+    pipeline_stack = (
+        CICDPipelineStack(
+            cdk_app,
+            id="dummy-pipeline",
+            environment_id="dev",
+            pipeline_name="dummy-pipeline",
+        )
+        .add_source_action(repository_name="dummy-repository")
+        .add_synth_action()
+        .build()
+        .add_security_lint_stage()
+        .add_test_stage()
+        .add_stage("dev", DevStage(cdk_app, "dev"))
+        .synth()
+    )
+    template = Template.from_stack(pipeline_stack)
+    # Check if synthesized pipeline contains source, synth, self-update, and app stage
+    template.has_resource_properties(
+        "AWS::CodePipeline::Pipeline",
+        props={
+            "Stages": Match.array_with(
+                pattern=[
+                    Match.object_like(
+                        pattern={
+                            "Name": "Source",
+                            "Actions": Match.array_with(
+                                pattern=[
+                                    Match.object_like(
+                                        pattern={
+                                            "Name": "dummy-repository",
```

Comment:
I would either:
1. Just have the `test_cicd_pipeline_full` test and remove the simple one
2. Or just test the parts that we have not already tested in the `test_cicd_pipeline_simple`, in this case the Security and Tests phases

Otherwise I just feel that we are duplicating code between the two tests if that makes sense?

Reply:
resolved in: https://github.com/awslabs/aws-ddk/pull/44